lib/instrumentation/modules/ioredis.js

```diff
@@ -1,81 +1,69 @@
 'use strict'
 
-var semver = require('semver')
+// Instrumentation of the 'ioredis' package:
+// https://github.com/luin/ioredis
+// https://github.com/luin/ioredis/blob/master/API.md
 
-var shimmer = require('../shimmer')
+const semver = require('semver')
 
-var spanSym = Symbol('elasticAPMSpan')
+const constants = require('../../constants')
+const { getDBDestination } = require('../context')
+const shimmer = require('../shimmer')
+
+const TYPE = 'cache'
+const SUBTYPE = 'redis'
+const hasIoredisSpanSym = Symbol('ElasticAPMHasIoredisSpan')
 
 module.exports = function (ioredis, agent, { version, enabled }) {
+  if (!enabled) {
+    return ioredis
+  }
   if (!semver.satisfies(version, '>=2.0.0 <5.0.0')) {
     agent.logger.debug('ioredis version %s not supported - aborting...', version)
     return ioredis
   }
 
-  agent.logger.debug('shimming ioredis.Command.prototype.initPromise')
-  shimmer.wrap(ioredis.Command && ioredis.Command.prototype, 'initPromise', wrapInitPromise)
-
-  if (!enabled) return ioredis
```

Comment:
REVIEW NOTE: This update of the ioredis instrumentation no longer wraps `Command.prototype.initPromise`; it still wraps `Redis.prototype.sendCommand`. The only utility of having wrapped `initPromise` before was something about "continuation patches" that may have been necessary in the past. This is why the previous instrumentation still wrapped `initPromise` even if `enabled=false` (from `disableInstrumentations` use). I believe this is no longer necessary after the run-context work. See https://github.com/elastic/apm-agent-nodejs/pull/2444 (a draft PR where I intend to remove this "half-disabling" of instrumentations) for some more discussion.

Reply:
> I believe this is no longer necessary after the run-context work.

Does remove the `initPromise` wrapping have any effect on the spans/span-tree if someone's using the `patch-async` /`asyncHooks:false`?  If so are you OK with changing that functionality and/or should we mention it somewhere?

Reply:
> Does remove the initPromise wrapping have any effect on the spans/span-tree if someone's using the patch-async /asyncHooks:false?

No, I don't believe it has any effect. The removed wrappedInitPromise didn't do any promise-y context propagation or binding. My quoting of "continuation patches" was from this single comment on the change that added the "enabled" flag to instrumentations: https://github.com/elastic/apm-agent-nodejs/pull/353#issuecomment-392212569   It provides no more details.

The test suite passes with `asyncHooks: false`. Also the parent/child output from `ELASTIC_APM_ASYNC_HOOKS=false node examples/trace-ioredis.js` is as expected:

```
    transaction 035eaa "t1"
    `- span 46aff5 "SET"
    `- span 4824ad "GET"
    `- span 77b874 "GET"
    `- span 0d4192 "MULTI"
    `- span 537be3 "SET"
    `- span 35b9c7 "GET"
    `- span 3a4fa9 "EXEC"
    `- span 6610bb "HSET"
    `- span 21c383 "GET"
      `- error 579426 "WRONGTYPE Operation against a key holding the wrong kind of value"
```

Reply:
👍